webapp/channels/src/utils/path.ts

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+export const ID_PATH_PATTERN = '[a-z0-9]{26}';
+
+// This should cover:
+// - Team name (lowercase english characters, numbers or -)
+// - Two ids separated by __ (userID__userID)
+export const TEAM_NAME_PATH_PATTERN = '[a-z0-9\\-_]+';
+
+// This should cover:
+// - Channel name
+// - Channel ID
+// - Group Channel Name (40 length UID)
+// - DM Name (userID__userID)
+// - Username prefixed by a @
+// - User ID
+// - Email
```

Comment:
Where do we use emails with this? This doesn't seem to cover even the most common email address formats like `juho.nurminen+mattermost@example.com`. Should it?

Reply:
I would love input of @hmhealey on this, but according to the code, (and testing around that) you can open a DM by having a url like `/teamName/channels/userEmail` (if I remember correctly).

What I am not sure is whether we ever expose that kind of link anywhere, and if it is something we want to fully support, or support at all.

The reality is that emails are hard to deal with regular expressions, and we didn't want to go with the blacklisting approach, so not sure how to deal with this properly.

Reply:
I honestly don't remember when that was added. I think that and the ability to visit a DM by the user's ID might've been a feature for Shift, but I don't think they were ever documented or anything. I think Calls or something similar visits a DM by user ID, but I don't know of anything we have that accesses DMs by email.

Because I don't think anyone actually uses it, I'd be fine risking breaking that in certain cases. Since it's so hard to actually validate emails, I don't think it's worth trying to do that perfectly if we don't know people are actually using it